File: programmers/155652.js
const ALPHABET = [
  'a',
  'b',
  'c',
  'd',
  'e',
  'f',
  'g',
  'h',
  'i',
  'j',
  'k',
  'l',
  'm',
  'n',
  'o',
  'p',
  'q',
  'r',
  's',
  't',
  'u',
  'v',
  'w',
  'x',
  'y',
  'z',
];

function solution(s, skip, index) {
  const alphabetSet = new Set(ALPHABET);
  for (const target of skip) {
    alphabetSet.delete(target);
  }
  const validAlphabet = [...alphabetSet.keys()];

  const encodingMap = new Map();
  validAlphabet.forEach((v, i) => {
    const value = (i + index) % validAlphabet.length;

    encodingMap.set(v, validAlphabet[value]);
  });

  return s
    .split('')
    .map(v => encodingMap.get(v))
    .join('');
}

console.log(solution('aukks', 'wbqd', 5));
console.log(solution('abcde', 'b', 3));

File: programmers/155652.py
def solution(s: str, skip: str, index: int) -> str:
    answer = ""
    skip_set = set([s for s in skip])

    string = "abcdefghijklmnopqrstuvwxyz"
    filtered_string = ""

    for c in string:
        if c in skip_set:
            continue
        else:
            filtered_string += c

    chr_dict = {}
    for i, c in enumerate(filtered_string):
        chr_dict[c] = i

    for c in s:
        answer += filtered_string[(chr_dict[c] + index) % len(filtered_string)]

    return answer


print(solution("aukks", "wbqd", 5))
